tests/unit/test_dilithium.cpp

```diff
@@ -0,0 +1,1260 @@
+                                        "I am exactly of size 32, true!!!"),
+                        std::make_tuple(AsymmetricKey::KeyTypes::DILITHIUM3,
+                                        "I am exactly of size 32, true!!!"),
+                        std::make_tuple(AsymmetricKey::KeyTypes::DILITHIUM5,
+                                        "I am exactly of size 32, true!!!")));
+
+void signAndVerify(const DilithiumAsymmetricPrivateKey &privKey, const std::string &messageStr)
+{
+    std::vector<uint8_t> message(messageStr.begin(), messageStr.end());
+    auto ctx = DilithiumSigningCtx(privKey);
+    auto res = ctx.signMessage(message);
+
+    auto pub_key_raw = privKey._internal()->getPublicKey();
+    auto pub_key = DilithiumAsymmetricPublicKey(std::make_shared<DilithiumKeyImpl>(pub_key_raw));
+
+    auto a = pub_key._internal()->getPublicKey().getKeyData();
+    auto b = pub_key_raw.getKeyData();
+    // test getter for public Key
+    EXPECT_EQ(pub_key._internal()->getPublicKey().getKeyData(), pub_key_raw.getKeyData());
+
+    auto verify_ctx = DilithiumVerificationCtx(pub_key);
+    verify_ctx.verifyMessage(res, message);
+}
+
+/* Test the KeySpec and the generation of keys that way */
+TEST_P(DilithiumSunnyDayTest, ValidKeyTypesAndMessages)
+{
+    auto spec = DilithiumSpec(std::get<0>(GetParam()));
+    auto messageStr = std::get<1>(GetParam());
+
+    auto priv_key = spec.generate();
+
+    auto priv_key_ctx = DilithiumAsymmetricKeypair(priv_key._internal());
+    // test getType function
+    EXPECT_EQ(priv_key.getType(), std::get<0>(GetParam()));
+    signAndVerify(priv_key_ctx, messageStr);
+}
+
+class DilithiumInvalidKeyTypesTest : public testing::TestWithParam<AsymmetricKey::KeyTypes>
+{
+};
+
+INSTANTIATE_TEST_SUITE_P(
+        InvalidKeyTypes,
+        DilithiumInvalidKeyTypesTest,
+        testing::Values(AsymmetricKey::KeyTypes::ECC,
+                        AsymmetricKey::KeyTypes::RSA,
+                        static_cast<AsymmetricKey::KeyTypes>(std::numeric_limits<int>::max())));
+
+TEST_P(DilithiumInvalidKeyTypesTest, InvalidKeyTypes)
+{
+    EXPECT_THROW(DilithiumSpec(GetParam()).generate(), MoCOCrWException);
+}
+
+TEST(DilithiumTest, CheckPrivateKeyGeneration)
+{
+    auto spec = DilithiumSpec(AsymmetricKey::KeyTypes::DILITHIUM3);
+    auto priv_key_ctx = DilithiumAsymmetricKeypair::generate(spec);
+    std::string messageStr = "You better believe I am loaded with both!";
+    signAndVerify(priv_key_ctx, messageStr);
+}
+
+struct BackendTestData
+{
+    std::string privateKeyDer;
+    std::string privateKeyRaw;
+    std::string publicKeyDer;
+    std::string publicKeyRaw;
+    std::string message;
+    std::string signature;
+};
+
+class DilithiumBackendDataTest : public testing::TestWithParam<BackendTestData>
+{
+};
+
+INSTANTIATE_TEST_SUITE_P(
+        BackendData,
+        DilithiumBackendDataTest,
+        testing::Values(
+                BackendTestData{
+
+                        // DILITHIUM 3
```

Comment:
Maybe just put large data into files? I think we already have some helpers to load data from files for other of the tests somewhere?